.github/workflows/build_doc.yml

```diff
@@ -24,10 +24,12 @@ jobs:
       run: |
         sudo apt-get -qqy update
         sudo apt-get -qy install texlive-latex-extra texlive-pstricks texlive-science
+        sudo apt install chktex
 
     - name: make
       run: |
         make
+        make lint
```

Comment:
I would suggest you to separate lint checking from generating pdf. For example: https://github.com/solvcon/mmnote/commit/b8f014c171e146691358ae49708f41e6f5749097